server/src/test/java/org/elasticsearch/snapshots/BlobStoreFormatTests.java

```diff
@@ -163,79 +148,6 @@ public void testBlobCorruption() throws IOException {
         }
     }
 
-    public void testAtomicWrite() throws Exception {
-        final BlobStore blobStore = createTestBlobStore();
-        final BlobContainer blobContainer = blobStore.blobContainer(BlobPath.cleanPath());
-        String testString = randomAlphaOfLength(randomInt(10000));
-        final CountDownLatch block = new CountDownLatch(1);
-        final CountDownLatch unblock = new CountDownLatch(1);
-        final BlobObj blobObj = new BlobObj(testString) {
-            @Override
-            public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
-                super.toXContent(builder, params);
-                // Block before finishing writing
-                try {
-                    block.countDown();
-                    unblock.await(5, TimeUnit.SECONDS);
-                } catch (InterruptedException ex) {
-                    Thread.currentThread().interrupt();
-                }
-                return builder;
-            }
-        };
-        final ChecksumBlobStoreFormat<BlobObj> checksumFormat = new ChecksumBlobStoreFormat<>(BLOB_CODEC, "%s", BlobObj::fromXContent,
-            xContentRegistry(), randomBoolean());
-        ExecutorService threadPool = Executors.newFixedThreadPool(1);
-        try {
-            Future<Void> future = threadPool.submit(new Callable<Void>() {
-                @Override
-                public Void call() throws Exception {
-                    checksumFormat.writeAtomic(blobObj, blobContainer, "test-blob");
-                    return null;
-                }
-            });
-            // signalling
-            block.await(5, TimeUnit.SECONDS);
-            assertFalse(blobContainer.blobExists("test-blob"));
-            unblock.countDown();
-            future.get();
-            assertTrue(blobContainer.blobExists("test-blob"));
-        } finally {
-            threadPool.shutdown();
-        }
-    }
-
-    public void testAtomicWriteFailures() throws Exception {
```

Comment:
This test and the other removed test were both fairly pointless to begin with. We never did any kind of streaming write in `ChecksumBlobStoreFormat`, so blocking in `toXContent` doesn't really test anything about atomic writes and the same tests would pass for non-atomic writes just the same. 
I simplified the code in the write path of `ChecksumBlobStoreFormat` a bit to make it clearer that we always serialize first, then physically write.